tfx/experimental/pipeline_testing/pipeline_recorder.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2020 Google LLC. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Recording pipeline from MLMD metadata."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from absl import app
+from absl import flags
+from collections import defaultdict
+from distutils.dir_util import copy_tree
+from ml_metadata.proto import metadata_store_pb2
+import os
+from tfx.orchestration import metadata
+
+FLAGS = flags.FLAGS
+
+default_pipeline_dir = os.path.join(os.environ['HOME'],
+                                    "tfx/pipelines/chicago_taxi_beam/")
+default_record_dir = os.path.join(os.environ['HOME'],
+                                  'tfx/tfx/experimental/pipeline_testing/',
+                                  'examples/chicago_taxi_pipeline/testdata')
+default_metadata_dir = os.path.join(os.environ['HOME'],
+                                    'tfx/tfx/experimental/pipeline_testing/',
+                                    'metadata.db')
+
+flags.DEFINE_string('pipeline_dir', default_pipeline_dir, 'Path to pipeline')
+flags.DEFINE_string('record_dir', default_record_dir, 'Path to record')
+flags.DEFINE_string('metadata_dir', default_metadata_dir, 'Path to metadata')
+flags.DEFINE_string('run_id', None, 'Pipeline Run Id')
+
+def main(unused_argv):
+  run_id = FLAGS.run_id
+  metadata_dir = FLAGS.metadata_dir
+  metadata_config = metadata.sqlite_metadata_connection_config(metadata_dir)
+  with metadata.Metadata(metadata_config) as m:
+    if not run_id:
```

Comment:
Please use `run_id is None` to check `None`. https://google.github.io/styleguide/pyguide.html#2144-decision